pkg/cacheutil/memcached_client.go

```diff
@@ -492,6 +473,27 @@ func (c *memcachedClient) getMultiSingle(ctx context.Context, keys []string) (it
 	return items, err
 }
 
+func (c *memcachedClient) handleError(op string, err error) {
+	var connErr *memcache.ConnectTimeoutError
+	var netErr net.Error
+	switch {
+	case errors.As(err, &connErr):
+		c.failures.WithLabelValues(op, reasonTimeout).Inc()
+	case errors.As(err, &netErr):
+		if netErr.Timeout() {
+			c.failures.WithLabelValues(op, reasonTimeout).Inc()
+		} else {
+			c.failures.WithLabelValues(op, reasonOther).Inc()
```

Comment:
I can introduce another label to cover `net.Error`. WDYT?

Reply:
Sure. Doesn't sound too bad in terms of cardinality. Is there a specific reason we would need it right now?

Reply:
Noop. That's why I went with other. I just need to catch the timeout errors.

Reply:
Agree having a dedicated label value would be great here.

Reply:
I don't feel this is problematic, it seems only helpful